pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -145,6 +148,23 @@ func getNetworkNameViaAPICall(svc *compute.Service, projectID string) (string, e
 	return networkList.Items[0].Name, nil
 }
 
+func getZonesForRegion(svc *compute.Service, projectID, region string) ([]string, error) {
+	listCall := svc.Zones.List(projectID)
+	res, err := listCall.Do()
+	if err != nil {
+		return nil, fmt.Errorf("unexpected response listing zones: %v", err)
+	}
+	zones := []string{}
+	for _, zone := range res.Items {
```

Comment:
nit: It seems that the API can do this filtering for you:

https://github.com/google/google-api-go-client/blob/master/compute/v1/compute-api.json#L13925

Less code is better :-)